cmake/FindHIP.cmake

```diff
@@ -529,6 +550,10 @@ endmacro()
 # HIP_ADD_LIBRARY
 ###############################################################################
 macro(HIP_ADD_LIBRARY hip_target)
+    # Add the include directories && compile definitions
+    list(APPEND HIP_HIPCC_INCLUDE_ARGS_USER "-I$<JOIN:$<TARGET_PROPERTY:${hip_target},INCLUDE_DIRECTORIES>, -I>")
+    list(APPEND HIP_HCC_FLAGS "-D$<JOIN:$<TARGET_PROPERTY:${hip_target},COMPILE_DEFINITIONS>, -D>")
```

Comment:
FindHIP supports hcc, nvcc and rudimentary support for hip-clang. So we need to target compile definitions for all paths i.e. atleast hcc and nvcc for now.
FindHIP does this by supporting ```HIP_HIPCC_FLAGS```, ```HIP_HCC_FLAGS``` and ```HIP_NVCC_FLAGS```. Since it is possible that the user may want different definitions to be enabled on each path, we cannot blindly add all ```COMPILE_DEFINITIONS``` to anyone specific compiler path. You should change this to ```HIP_HIPCC_FLAGS``` here since that targets all compiler paths.
It probably makes sense to have the same behavior in ```HIP_ADD_EXECUTABLE``` as well. So it is better to move these two statements to ```HIP_PREPARE_TARGET_COMMANDS```.

Reply:
@mangupta Great. So it looks like I'll have to extend my change to fix the NVCC path as well using HIP_HIPCC_FLAGS. I didn't catch that since I've only been using HIP/HCC, never the NVCC path which relies on marshaling within HIP.

I'll send an update!